examples/feature_engineering.py

```diff
@@ -0,0 +1,333 @@
+
+@task
+def read_csv(
+    rootdir: str, table_name: str, nrows: int = None, exclude_cols: List[str] = None
+):
+    """
+    Read the contents of a CSV into a Pandas DataFrame.
+
+    This is built specifically for the file structure of the Lahman database.
+
+    Args:
+        - rootdir (str): name of the top-level directory where the data is stored
+        - table_name (str): name of the data table to be retrieved
+        - nrows (int, optional): defaults to all rows; otherwise, number of rows to load
+        - exclude_cols (List[str], optional): columns to be excluded, by name
+
+    Returns:
+        - pd.DataFrame: the contents of the CSV
+    """
+    filepath = os.path.join("data", rootdir, "core", table_name + ".csv")
+    df = pd.read_csv(filepath, nrows=None)
+    return df.drop(exclude_cols, axis=1)
+
+
+@task
+def get_colnames(table: pd.DataFrame) -> List[str]:
+    """
+    Get column names from a DataFrame.
+    """
+    return list(table.columns)
+
+
+@task
+def get_cols(table: pd.DataFrame) -> List[np.ndarray]:
+    """
+    Get list of arrays corresponding to the columns of a DataFrame.
+    """
+    return list(table.values.T)
+
+
+# data manipulation tasks
+@task
+def impute(data: np.ndarray, replacement_dict: Dict[Any, Any]) -> np.ndarray:
+    """
+    Replace any instances of a set of values with corresponding new values.
+
+    The replacements are provided in a dictionary.
+    The keys are the values to be replaced,
+    and the values are the values to replace with.
+
+    Args:
+        - data (np.ndarray): data in which the replacements should be made
+        - replacement_dict (Dict[Any, Any]): what to replace, and what to replace it with
+
+    Returns:
+        - np.ndarray: data with values replaced
+    """
+    for old, new in replacement_dict.items():
+        if np.isnan(old):
+            data[pd.isnull(data)] = new
+        else:
+            data[data == old] = new
+    return data
+
+
+@task
+def cast(data: np.ndarray, new_type: type) -> np.ndarray:
+    """
+    Cast all the values of an array to a new data type.
+
+    Args:
+        - data (np.ndarray): data to be cast
+        - new_type (type): type to be cast to
+
+    Returns:
+        - np.ndarray: the data in its new type
+    """
+    return data.astype(new_type)
+
+
+@task
+def get(values: List[np.ndarray], keys: List[str], key: str) -> np.ndarray:
+    """
+    Retrieve the value associated with a key given a list of keys and list of values.
+
+    Essentially a utility to make use of dicts that have been broken into
+    a list of keys and a list of values.
+
+    Args:
+        - values (List[np.ndarray]): values associated with keys
+        - keys (List[str]): keys to choose from
+        - key (str): the chosen key
+
+    Returns:
+        - np.ndarray: the value associated with the chosen key
+    """
+    return values[keys.index(key)]
+
+
+# using a specific number of arrays since Prefect doesn't support variable positional args
```

Comment:
for this, you could alternatively refactor these two tasks into one with a signature of:
```python
def concatenate(arrays: List[np.ndarray]) -> np.ndarray
```